Detectors/MUON/MCH/DevIO/Digits/digits-random-generator-workflow.cxx

```diff
@@ -0,0 +1,210 @@
+// Copyright CERN and copyright holders of ALICE O2. This software is
+// distributed under the terms of the GNU General Public License v3 (GPL
+// Version 3), copied verbatim in the file "COPYING".
+//
+// See http://alice-o2.web.cern.ch/license for full licensing information.
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+#include "CommonDataFormat/InteractionRecord.h"
+#include "DataFormatsMCH/Digit.h"
+#include "DataFormatsMCH/ROFRecord.h"
+#include "DigitIOBaseTask.h"
+#include "Framework/ConfigParamRegistry.h"
+#include "Framework/ConfigParamSpec.h"
+#include "Framework/ControlService.h"
+#include "Framework/DataProcessorSpec.h"
+#include "Framework/DataSpecUtils.h"
+#include "Framework/Lifetime.h"
+#include "Framework/Logger.h"
+#include "Framework/Output.h"
+#include "Framework/Task.h"
+#include "Framework/WorkflowSpec.h"
+#include "MCHMappingInterface/Segmentation.h"
+#include "ProgOptions.h"
+#include <array>
+#include <chrono>
+#include <cstring>
+#include <fmt/format.h>
+#include <fstream>
+#include <iostream>
+#include <random>
+#include <stdexcept>
+#include <vector>
+
+/**
+ * This executable generate random MCH digits and their associated ROFRecords.
+ *
+ */
+
+using namespace o2::framework;
+
+using namespace o2::mch;
+
+namespace
+{
+std::array<int, 156> getDeIds()
```

Comment:
just for the record: here and in the next method you are sending a copy of the same array filled at the 1st call. Is this what you want?

Reply:
Well, in both cases what I want to avoid is repeated calls to the segmentation, because I know (i.e. have measured) those are costly. After that, to copy or not an array of integer would be a matter of micro-optimization I would say. Unless that'd be measured to be a problem I would not care too much about it (in particular in this workflow which is by definition not meant for production)